lib/Doctrine/ORM/Query/Filter/SQLFilter.php

```diff
@@ -102,6 +102,27 @@ abstract class SQLFilter
         return isset($this->parameters[$name]);
     }
 
+    /**
+     * Return a unique hash base on filter class and his parametters
+     *
+     * @return string
+     */
+    final public function getHash()
+    {
+        $parameterCount = count($this->parameters);
+        $hash           = get_class($this).$parameterCount;
+        foreach ($this->parameters as $name => $param) {
+            if (is_object($param['value'])) {
+                $valueHash = spl_object_hash($param['value']);
```

Comment:
`spl_object_hash` is not reliable within and across requests. You need to serialize the parameter. the API of `setParameter` only allows stirngs anyways, it is not enforced though. But that should allow us to do this.

Reply:
How about changing this method to
```php
final public function getHash()
{
        return self::class.$this;
}
```
I don't think the serialize() will have a big performance impact here.